sklearn/decomposition/_pca.py

```diff
@@ -636,12 +649,12 @@ def _fit_truncated(self, X, n_components, svd_solver):
         # Workaround in-place variance calculation since at the time numpy
         # did not have a way to calculate variance in-place.
         N = X.shape[0] - 1
-        np.square(X, out=X)
-        np.sum(X, axis=0, out=X[0])
+        xp.square(X, out=X)
+        xp.sum(X, axis=0, out=X[0])
         total_var = (X[0] / N).sum()
 
         self.explained_variance_ratio_ = self.explained_variance_ / total_var
-        self.singular_values_ = S.copy()  # Store the singular values.
+        self.singular_values_ = xp.asarray(S)  # Store the singular values.
```

Comment:
If you want to guarantee a copy here you should use `asarray(S, copy=True)`.

Reply:
Sure! Added.